File: StudyBuddy.Model/Model/NotificationUserMetadata.cs
using System;

namespace StudyBuddy.Model
{
    public class NotificationUserMetadata : Entity
    {
        public int NotificationId { get; set; } = 0;
        public int OwnerId { get; set; } = 0;
        public bool? Liked { get; set; }
        public bool? Seen { get; set; }
        public bool? Shared { get; set; }
        public DateTime Updated { get; set; } = DateTime.Now;
    }
}

File: StudyBuddy.Backend/src/app/model/notification-user-metadata.ts
export class NotificationUserMetadata {
    id: number;
    notificationId: number;
    ownerId: number;
    liked: boolean | null;
    seen: boolean | null;
    shared: boolean | null;
    created: string;
    updated: string;
    

    static fromApi(obj): NotificationUserMetadata {
        return Object.assign(new NotificationUserMetadata(), obj)
    }

    toApi() {
        return this;
    }
}